website/profile/utils.py

```diff
@@ -45,7 +49,7 @@ def serialize_user(user, node=None, admin=False, full=False, is_profile=False):
             }
         else:
             flags = {
-                'visible': node.contributor_set.filter(user=user, visible=True).exists(),
+                'visible': getattr(contrib, 'visible', node.contributor_set.filter(user=user, visible=True).exists()),
```

Comment:
Yeah, based on the change above if we get a contributor instance as the user variable then it will reassign contrib to the contributor and set user to the user. Once we're here if it's not a contributor instance then it won't have a visible and it'll do the extra query but if it is a contributor then it skips that query.